src/scenarios/billing/ListSubscriptionPlansForUser/ListSubscriptionPlansForUser.tsx

```diff
@@ -0,0 +1,211 @@
+        fetchPolicy: 'no-cache',
+      });
+
+      if (activeSubResult.errors) {
+        logger.error(
+          `method [${listSubscriptionPlansForUser.name}]`,
+          'output:',
+          activeSubResult.errors,
+        );
+      } else {
+        const subscriptions = activeSubResult.data.subscriptions.nodes;
+        if (Array.isArray(subscriptions) && subscriptions.length > 0) {
+          activeSubscription = subscriptions[0];
+
+          if (subscriptionPlanFilter) {
+            subscriptionPlanFilter = {
+              ...subscriptionPlanFilter,
+              id: { notEqualTo: activeSubscription.subscriptionPlan.id },
+            };
+          } else {
+            subscriptionPlanFilter = {
+              id: { notEqualTo: activeSubscription.subscriptionPlan.id },
+            };
+          }
+        }
+      }
+
+      const availableSubResult = await apolloClient.query({
+        query: getSubscriptionPlansQuery,
+        variables: {
+          subscriptionPlanFilter,
+          paymentPlanPriceFilter,
+        },
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      if (availableSubResult.errors) {
+        logger.error(
+          `method [${listSubscriptionPlansForUser.name}]`,
+          'output:',
+          availableSubResult.errors,
+        );
+      } else {
+        if (activeSubscription) {
+          logger.log(
+            `method [${listSubscriptionPlansForUser.name}]`,
+            'output: The user has an active subscription: ',
+            activeSubscription,
+            'Other available subscription plans: ',
+            availableSubResult.data.subscriptionPlans.nodes,
+          );
+        } else {
+          logger.log(
+            `method [${listSubscriptionPlansForUser.name}]`,
+            'output: The user does not have an active subscription.',
+            'Available subscription plans: ',
+            availableSubResult.data.subscriptionPlans.nodes,
+          );
+        }
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        logger.error(
+          `method [${listSubscriptionPlansForUser.name}]`,
+          'output:',
+          error.message,
+        );
+      }
+    }
+  };
+
+  return (
+    <>
+      <Segment basic>
+        <Header size="huge">List Subscription Plans for a User</Header>
+        <Header size="small">
+          Required Services:
+          <Label>user-service</Label>
+          <Label>billing-service & monetization-service</Label>
+        </Header>
+
+        <Divider />
+
+        <Container fluid>
+          <p>
+            This scenario demonstrates how to retrieve and show a list of
+            subscription plans for a signed-in user. If the user has an active
+            subscription, it will be listed with corresponding details.
+            Otherwise all available subscription plans will be listed.
```

Comment:
This scenario demonstrates how to retrieve the current subscription details of the user (if any), and also will list the other available subscription plans the user can subscribe to.

Reply:
Done